lib/Sema/CSDiag.cpp

```diff
@@ -455,9 +456,91 @@ tryDiagnoseTrailingClosureAmbiguity(TypeChecker &tc,
   return true;
 }
 
-static bool diagnoseAmbiguity(ConstraintSystem &cs,
-                              ArrayRef<Solution> solutions,
-                              Expr *expr) {
+bool ConstraintSystem::diagnoseAmbiguityWithFixes(
```

Comment:
Maybe move ambiguity diagnosis out of CSDiag then, if it doesn’t involve re-checking? I’d like to let CSDiag atrophy so we can delete it wholesale once we’re ready.

Reply:
I think it makes sense, I was thinking that after this is done to move both `diagnoseAmbiguity*` methods to `ConstraintSystem.cpp`, didn't want to do it in the PR though to minimize changes, if I do it separately it could be NFC.